#3338 details for part submission

GauriRajesh733 opened the pull request:
## Changes

Added PartSubmissionDetails.tsx component in Components folder under PartPage.  Example PartSubmission was manually added to PartPage.tsx to test what it looked like.

## Notes

Slightly changed alignment of existing components from 100% to 50% in PartPage.tsx to test what PartSubmissionDetails component looked like.

## Test Cases

None

## Screenshots

![partsubmissiondetails2](https://github.com/user-attachments/assets/3ba4c03d-0fdb-4e73-863e-a0794ac6cbcd)
![partsubmissiondetails1](https://github.com/user-attachments/assets/fab311e2-af0e-4f50-9d1f-88f41d3979a7)


## To Do

None

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #3338

GauriRajesh733 committed:
Update PartPage.tsx

#3338: added part submission details component

GauriRajesh733 committed:
Create PartSubmissionDetails.tsx

#3338: added front end component for project submission details

gcooper407 committed:
#3338: added sample submission details to help dev get started on testing component

GauriRajesh733 committed:
Update PartSubmissionDetails.tsx

#3338: changed alignment of components in part submission

GauriRajesh733 committed:
Update PartSubmissionDetails.tsx

#3338: testing alignment of components in part submission

GauriRajesh733 committed:
Update PartSubmissionDetails.tsx

#3338: corrected alignment issues with part submission component

GauriRajesh733 committed:
Update PartPage.tsx

#3338: changed alignment of partpage components

chpy04 commented on src/frontend/src/pages/PartPage/Components/PartSubmissionDetails.tsx:
```diff
@@ -0,0 +1,34 @@
+const PartSubmissionDetails = ({ submission }: PartSubmissionProps) => {
+  return (
+    <Box display="left" alignItems="center" width="100%" sx={{ flexDirection: 'column'}}>
+      <Typography sx = {{fontWeight: "normal", marginBottom: '4%'}} variant="h5">Details for Submission #{submission.partSubmissionId}:</Typography>
```
This should be `Details for ${submission.name}`, in the context of the epic, 'submission #1' is the name of the submission

chpy04 commented on src/frontend/src/pages/PartPage/Components/PartSubmissionDetails.tsx:
```diff
@@ -0,0 +1,34 @@
+      <Typography sx = {{fontWeight: "normal", marginBottom: '4%'}} variant="h5">Details for Submission #{submission.partSubmissionId}:</Typography>
+
+      <Typography variant="body1" sx = {{marginBottom: '4%'}}>
+        Uploader:
```
You'll want to use a template string here, which makes it so you don't have to do the + ' ' + ..., and instead you can just have something like `Uploader: ${submission.userCreated.firstName} ${submission.userCreated.lastName}`

chpy04 commented on src/frontend/src/pages/PartPage/Components/PartSubmissionDetails.tsx:
```diff
@@ -0,0 +1,34 @@
+      <Typography variant="body1" sx = {{marginBottom: '4%'}}>
+        Reviewer Notes:{' '}
+        {submission.reviews.length !== 0
+          ? submission.reviews.map((review) => review.notes).join('  ')
```
Can you join these on newLines so they show up ontop of one another. Also, can you put an example of there being multiple review notes in your PR

chpy04 commented on src/frontend/src/pages/PartPage/Components/PartSubmissionDetails.tsx:
```diff
@@ -0,0 +1,34 @@
+
+const PartSubmissionDetails = ({ submission }: PartSubmissionProps) => {
+  return (
+    <Box display="left" alignItems="center" width="100%" sx={{ flexDirection: 'column'}}>
```
Instead of using a box and having to put the marginBottom on every typography, you should be able to use a Stack and just set the spacing field once

chpy04 reviewed (changes requested):
Looks awesome! just needs a few changes and then should be good to go!

GauriRajesh733 committed:
#3338: corrections after first pr

GauriRajesh733 committed:
#3338: tried adding mock examples

running into some errors on PartPage when adding multiple mock review examples.  it's saying that the popUp types are incompatible

GauriRajesh733 committed:
#3338: added mock review examples

example with multiple part reviews to test frontend component

GauriRajesh733 committed:
#3338: fixed alignment of component

GauriRajesh733 commented:
Added example with multiple reviews and made corrections!
![image](https://github.com/user-attachments/assets/50400af1-3b6a-4d0c-aa0f-30487ddf8316)

chpy04 reviewed (approved):
LGTM!

walker-sean reviewed (changes requested):
This looks great so far! Could we make everything go on a line below the heading (and bold the heading or otherwise make it stand out)?

I think especially for the notes it would look a lot cleaner. Also the reviews could be bullets which would help to more easily parse reviews

walker-sean commented on src/frontend/src/pages/PartPage/PartPage.tsx:
```diff
@@ -63,18 +65,39 @@ const PartPage: React.FC = () => {
                 }}
               >
-                <Typography>History</Typography>
+                {/*Add part submission details here*/}
```
Also, this was left in

GauriRajesh733 committed:
#3338 - edits for readability and space

GauriRajesh733 committed:
#3338 - mock examples

multiple part submission reviews

GauriRajesh733 committed:
#3338 - linting fix
